### ISSUE TITLE
error_pages with relative directory does not with php files

### ISSUE THREAD

**Issue by tkauppin** [civetweb/civetweb#671]
The following configuration does not work properly:
* document_root ..\public_html (works OK, also with PHP files)
* error_pages ..\public_html\errors\ (HTML files works ok, PHP cannot find the input file)
Probably the ".." in the directory name causes the problems.

if using a regular relative directory, it works fine, e.g. document root=. and error_pages=errors\

**Comment by bel2125** [comment]
Yes, ".." may cause problems. In some cases (e.g. in URLs) the ".." is explicitly forbidden - you are not allowed to move out of a directory, only deeper into it.
Internally civetweb.c works with absolute directories - if you use the precompiled executable (or main.c), this will modify parameters like "dokument_root" to full paths (see [main.c](https://github.com/civetweb/civetweb/blob/c47476e289e3bb4543ad9ab2ddf1c286ebbe97ab/src/main.c#L999)). Furthermore, it depends on if the server is run on Windows or Linux (these systems behave a little different). On Windows, I would reccomend to use a start bat (or cmd) file in some directory and put binaries, certificates, document root, working directory and off root contents (databases, ...) - in five different subdirectories below the start bat. On Linux, you maybe want to use the default Linux file hierarchy standard, but then I strongly reccomend to only use full absolute paths (or you ignore the file hierarchy and do it like in Windows in your home, opt or whatever directory).

**Comment by tkauppin** [comment]
I understand the point. I'm using the CivetWeb only for Windows, development purposes for my php-based website before the deployment to the Linux env. Where the error pages are configured by Apache. 

The reason for rising the issue is simply this:
- html error pages works fine
- ordinary php pages are handled OK with similar kind of relative doc_root
- Looks like the php page filename is passed incorrectly to the PHP

If the error pages should be located inside the doc_root, then the following logic would work better:
- doc_root=../public_html, error_pages=errors/  => full path to error page = doc_root + errors + error404.php (only if relative paths are used)

**Comment by bel2125** [comment]
CivetWeb does not require the error pages to be within the doc_root.
I tend to not put it there, but you certainly can if you specify the full path.
Directory structures are always a matter of taste as well - but even if a new approach of interpreting directory configurations might me a little more comprehensible, changing the definition would cause widespread compatibility issues. Multiple applications work with the current definition for many years.
What's definitely possible is a better documentation on the use of directories and corresponding configuration values.

I just had a look - it seems I already explained my recommended directory structure for Windows at least 3 times - but I never wrote it to a documentations since it's probably just my "taste of directory structures" and other solutions are perfectly valid as well. Anyway, these are the 3 comments I found (improved from year to year):
- from 2015-01-06: https://github.com/civetweb/civetweb/issues/56#issuecomment-68888367
- from 2016-04-20: https://github.com/civetweb/civetweb/issues/299#issuecomment-212534383
- from 2017-01-17: https://github.com/civetweb/civetweb/issues/400#issuecomment-273299652
- additional recommendation: use whatever structure you find your way, but avoid ".."
